lib/publiccloud/k8s_provider.pm

```diff
@@ -0,0 +1,78 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for amazon connection and authentication
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>, qa-c team <qa-c@suse.de>
+
+package publiccloud::k8s_provider;
+use Mojo::Base -base;
+use testapi;
+use mmapi 'get_current_job_id';
+
+has key_id => undef;
+has key_secret => undef;
+has region => undef;
+has resource_name => sub { get_var('PUBLIC_CLOUD_RESOURCE_NAME', 'openqa-vm') };
+has container_registry => undef;
+has provider_client => undef;
+has service => undef;
+
+sub init {
+    my ($self, $service) = @_;
+
+    die('The service must be specified') if (!$service);
```

Comment:
`service` was introduced as workaround to keep k8s logic together with instance related at same flow . now after split do we really need it ? Below you actually use `$provider` anyway so maybe we can drop this variable ?

Reply:
Service is needed to be passthrough to  each clients (aws_client...) so I am removing the variable from this class because is not needed to be stored

Reply:
Done